app/views/handlers/confirm_email.py

```diff
@@ -153,17 +148,17 @@ def _publish_fulfilment_request(self):
 @dataclass
 class ConfirmationEmailFulfilmentRequest(FulfilmentRequest):
     email_address: str
-    session_data: SessionData
+    metadata: dict[str, Any]
     schema: QuestionnaireSchema
 
     def _payload(self) -> Mapping:
         return {
             "fulfilmentRequest": {
                 "email_address": self.email_address,
-                "display_address": self.session_data.display_address,
+                "display_address": self.metadata["display_address"],
```

Comment:
I would use `.get()` as `display_address` would only exist in the census flow. If we make it graceful, we could reuse for others even if those didn't have `display_address`.